src/sinks/humio/metrics.rs

```diff
@@ -69,7 +70,7 @@ impl GenerateConfig for HumioMetricsConfig {
 #[typetag::serde(name = "humio_metrics")]
 impl SinkConfig for HumioMetricsConfig {
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
-        let mut transform = self.transform.clone().build().await?;
+        let mut transform = self.transform.clone().build().await?.wrap_emit();
```

Comment:
I have a feeling we'll forget to do this in the future, especially as we start to expand our integrations. Is there a way to force the developer to explicitly choose? Such as a required argument in `build()` or something along those lines.

Please disregard if this doesn't make any sense 😄

Reply:
Good idea, but we can do even better. I think it would be possible to completely automate this, but I'll need to explore this idea a bit.

Reply:
We could detect build inside other build:
   * Will not catch builds outside of build method, but this shouldn't happen since errors of sub builds should be propagated through master build so this should lead contributors to build inside other build. (But this is a lot of shoulds) 
   * This can be done by wrapping build future to set thread local flag from which we can tell if this is main build or sub build.
   * Requires a layer of traits for 
         * `SourceConfig`
         * `TransformConfig`
         * `SinkConfig`
     in which we would modify built future to track the flag and wrap built component with `wrap_emit` or not depending on the flag.

cc. @lukesteensen

Reply:
Scratch that, once problematic cases like `new_relic_logs` are taken into consideration automatic detection will run into the same problems as #5174 .

That said, I think finding a way to force the developer to explicitly choose will be better served in a separate issue which can then build upon this PR.

Reply:
> That said, I think finding a way to force the developer to explicitly choose will be better served in a separate issue which can then build upon this PR.

I agree, this will allow us to progress in the meantime, and unblock other works that depend on this.